### ISSUE TITLE
Proposal: extend `no-magic-timeouts` to helper parameters (positional budgets and defaults)

### ISSUE THREAD

**Issue by neupenguin** [mskelton/eslint-plugin-playwright#512]
## The problem

`no-magic-timeouts` (#487, shipped in 2.12.0) reads `{ timeout: … }` options and `test.setTimeout(n)` / `testInfo.setTimeout(n)`. That covers direct Playwright calls, but in a suite with page objects and helpers most budgets never reach Playwright as an options literal. They are passed positionally into a wrapper, or written as the wrapper's default:

```ts
async function waitUntilVisible(locator: Locator, timeout = 15_000) {
  await expect(locator).toBeVisible({ timeout }) // named: not reported
}

await waitUntilVisible(dialog, 30_000)       // positional: not reported
await reportPage.open(2 * 60_000)            // into `budgetMs`: not reported
const settle = (ms = 400) => page.waitForTimeout(ms)
```

These are the same unnamed number the rule exists to catch; they have just moved one call away. After enabling `no-magic-timeouts` and driving it to zero on our suite, about sixty of these remained, found by grep, and nothing stops a new one.

## Proposal

Extend `no-magic-timeouts` with two opt-in checks (or ship them as a companion rule if that fits the plugin better):

1. **Defaults.** A numeric literal default on a binding whose name looks like a budget: `timeout = 15_000`, `{ budgetMs = 60_000 } = {}`, `(ms = 400) =>`. Syntax only, no type information needed.
2. **Positional arguments.** A numeric literal passed to a parameter whose name looks like a budget. The parameter name is resolved with the TypeScript checker (`getResolvedSignature` → `getParameters()`), so every helper and page-object method is covered without maintaining a list of function names. Skipped when type information is unavailable.

Both reuse the existing literal evaluation, so `2 * 60_000` is reported and anything containing an identifier is not.

### Options

- `parameterNames: string` — regex for budget-like names. Suggested default: `timeout|budget|^(ms|delay|wait|poll)(Ms)?$` (case-insensitive). Deliberately not any `…Ms`: `nowMs` and `closedAtMs` hold timestamps.
- `checkDefaults: boolean`, `checkArguments: boolean` — default `false`, so existing configs are unchanged.
- The existing `allow` list applies to both (`[0]` covers "disable" / "now").

### Not reported

- `waitForTimeout(n)` — a static wait, already the job of `no-wait-for-timeout`.
- `setTimeout(n)` — already handled by the current rule.
- `setInterval(fn, n)` and `toPass`'s `intervals` — sampling intervals, i.e. how often to look, not how long to wait.

## Prior art

We run this as a local rule today (TypeScript, typed `RuleTester` cases for named, composed, zero, timestamp-named, static-wait, `setTimeout` and `setInterval` as valid; positional, arithmetic, default, destructured-default and arrow-default as invalid). Happy to open a PR porting it, with tests and docs, if this direction works for you.
